### PR TITLE
Improve CHECK TABLE system query, support checking part

### PR DESCRIPTION
**Pull request by vdimir**
Resubmit PR #52745
### Changelog category (leave one):

- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
* Query`CHECK TABLE` has better performance and usability (sends progress updates, cancellable). Support checking particular part with `CHECK TABLE ... PART 'part_name'`

Changes from previous PR:
- Added `max_threads` where it was missing
- Use `max_threads = 1` if `check_query_single_value_result` is set to `0` and `max_threads` is not explicitly set in the query to support current behavior with deterministic result by default

### CODE REVIEW

File: src/Interpreters/InterpreterCheckQuery.cpp [hunk 1]
```diff
@@ -3,77 +3,211 @@
...
-        auto is_passed_column = block_structure[1].type->createColumn();
-        auto message_column = block_structure[2].type->createColumn();
+        auto processors = std::make_shared<Processors>();
+
+        std::vector<OutputPort *> worker_ports;
+
+        /// For verbose mode (`check_query_single_value_result = 0`)
+        /// if `max_threads` is not explicitly set, we will use single thread to have deterministic output order
+        const auto * settings_ast = typeid_cast<const ASTSetQuery *>(check.settings_ast.get());
+        bool use_max_threads = settings.check_query_single_value_result || (settings_ast && settings_ast->changes.tryGet("max_threads") != nullptr);
```
Comment (tavplubix) [inline src/Interpreters/InterpreterCheckQuery.cpp:174]:
```
SET max thread = 1;
CHECK TABLE mytable;  -- no settings_ast
```

File: src/Interpreters/InterpreterCheckQuery.cpp [hunk 2]
```diff
@@ -3,77 +3,211 @@
...
-        auto block_structure = getBlockStructure();
-        auto path_column = block_structure[0].type->createColumn();
-        auto is_passed_column = block_structure[1].type->createColumn();
-        auto message_column = block_structure[2].type->createColumn();
+        auto processors = std::make_shared<Processors>();
+
+        std::vector<OutputPort *> worker_ports;
+
+        /// For verbose mode (`check_query_single_value_result = 0`)
+        /// if `max_threads` is not explicitly set, we will use single thread to have deterministic output order
```
Comment (tavplubix) [inline src/Interpreters/InterpreterCheckQuery.cpp:172]: Maybe we need the opposite? Max threads by default and deterministic order for tests

**Review by tavplubix — APPROVED** [review]

File: src/Interpreters/InterpreterCheckQuery.cpp (hunk 2 above)
Comment (vdimir) [reply src/Interpreters/InterpreterCheckQuery.cpp]: Let's have one CI run with results shuffling https://github.com/ClickHouse/ClickHouse/pull/53404/commits/7b55c68130bd87a6c153dc2a9b7b55c91bf2fed3 to be sure that I didn't forget to add `max_threads = 1` in any test. Once it's passed, I'll remove the commit from this PR

Comment (vdimir) [reply src/Interpreters/InterpreterCheckQuery.cpp]: CI passed @ https://github.com/ClickHouse/ClickHouse/commit/1e068640c8f9912017612b9f40bacbb9029c4546, except an error in debug build. I don't think we need to rerun with randomization, release tests result is enough, I'm going to revert result randomization and fix debug build

**Comment by vdimir** [comment]
Another thing to add is query to check all tables in all databases. We can do it in this or another PR

**Comment by vdimir** [comment]
Another thing to add is query to check all tables in all databases. We can do it in this or another PR

**Review by tavplubix — APPROVED** [review]

**Pull request merged**